crev-data/src/proof/mod.rs

```diff
@@ -259,13 +259,11 @@ impl Proof {
                                 self.body += line;
                                 self.body += "\n";
                             }
+                        } else if is_signature_line(line) {
+                            self.stage = Stage::Signature;
                         } else {
-                            if is_signature_line(line) {
-                                self.stage = Stage::Signature;
-                            } else {
-                                self.body += line;
-                                self.body += "\n";
-                            }
+                            self.body += line;
```

Comment:
I actually disagree with this lint in clippy. Oftentimes it makes it less-clear what's going on, when there are multiple nested logical choices. By flattening them, some information is lost. But not a blocker.